Documentation or example for the allocator not working

The documentation appears to be out of date. If I follow the steps then, to build the allocator example, I get the following error:

```
% cargo build --example allocator         
   Compiling typenum v1.12.0
   Compiling version_check v0.9.2
   Compiling semver-parser v0.7.0
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling cortex-m v0.6.3
   Compiling syn v1.0.42
   Compiling stable_deref_trait v1.2.0
   Compiling vcell v0.1.2
   Compiling cortex-m-semihosting v0.3.5
   Compiling volatile-register v0.1.2
   Compiling bitfield v0.13.2
   Compiling cortex-m-rt v0.6.13
   Compiling test v0.1.0 (/Users/huntc/Desktop/test/test)
   Compiling r0 v0.2.2
   Compiling linked_list_allocator v0.6.6
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /Users/huntc/.cargo/registry/src/github.com-1ecc6299db9ec823/linked_list_allocator-0.6.6/src/lib.rs:1:1
  |
1 | #![feature(const_fn)]
  | ^^^^^^^^^^^^^^^^^^^^^
...
```

If I try with the nightly:

```
huntc@Christophers-MacBook-Pro test % cargo +nightly build --example allocator
   Compiling typenum v1.12.0
   Compiling semver-parser v0.7.0
   Compiling version_check v0.9.2
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling cortex-m v0.6.3
   Compiling syn v1.0.42
   Compiling stable_deref_trait v1.2.0
   Compiling vcell v0.1.2
   Compiling cortex-m-semihosting v0.3.5
   Compiling volatile-register v0.1.2
   Compiling cortex-m-rt v0.6.13
   Compiling bitfield v0.13.2
   Compiling linked_list_allocator v0.6.6
   Compiling test v0.1.0 (/Users/huntc/Desktop/test/test)
   Compiling r0 v0.2.2
error[E0432]: unresolved import `alloc::alloc::AllocErr`
  --> /Users/huntc/.cargo/registry/src/github.com-1ecc6299db9ec823/linked_list_allocator-0.6.6/src/lib.rs:14:30
   |
14 | use alloc::alloc::{AllocRef, AllocErr, Layout};
   |                              ^^^^^^^^
   |                              |
   |                              no `AllocErr` in `alloc`
   |                              help: a similar name exists in the module: `AllocError`
...
```

Any pointers no how to see that the allocator example is able to be built? Is the documentation/example out of date?

Thanks @thalesfragoso . I did that, but then receive:

```
% cargo build --example allocator
   Compiling linked_list_allocator v0.8.6
   Compiling typenum v1.12.0
   Compiling semver v0.9.0
   Compiling generic-array v0.14.4
   Compiling proc-macro2 v1.0.24
   Compiling cortex-m v0.6.3
   Compiling syn v1.0.42
   Compiling cortex-m-rt v0.6.13
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /Users/huntc/.cargo/registry/src/github.com-1ecc6299db9ec823/linked_list_allocator-0.8.6/src/lib.rs:1:1
  |
1 | #![feature(const_mut_refs)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0554`.
error: could not compile `linked_list_allocator`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ panic-halt = "0.2.0"
 # panic-itm = "0.4.1"
 
 # Uncomment for the allocator example.
-# alloc-cortex-m = "0.3.5"
+# alloc-cortex-m = "0.4.0"
 
 # Uncomment for the device example.
 # Update `memory.x`, set target to `thumbv7em-none-eabihf` in `.cargo/config`,

--- a/examples/allocator.rs
+++ b/examples/allocator.rs
@@ -9,7 +9,6 @@
 //!
 //! ---
 
-#![feature(alloc)]
 #![feature(alloc_error_handler)]
 #![no_main]
 #![no_std]
@@ -23,7 +22,7 @@ use core::alloc::Layout;
 use alloc_cortex_m::CortexMHeap;
 use cortex_m::asm;
 use cortex_m_rt::entry;
-use cortex_m_semihosting::hprintln;
+use cortex_m_semihosting::{hprintln, debug};
 
 // this is the allocator the application will use
 #[global_allocator]

